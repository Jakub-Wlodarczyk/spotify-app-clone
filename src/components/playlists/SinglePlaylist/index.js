import React from 'react';

class SinglePlaylist extends React.Component {
	constructor(props) {
		super(props);

		this.handlePlaylistCreation = this.handlePlaylistCreation.bind(this);
	}

	handlePlaylistCreation() {
		this.setState((prevState) => (
			{showPlaylistCreation: !prevState.showPlaylistCreation}
		));
	}

	render() {
		const { playlist } = this.props;

		return (
			<div className="single-song">
				<p><b>Playlist name:</b> {playlist.name}</p>
				<p><b>Songs:</b></p>
				{playlist.songs.length > 0 && playlist.songs.map(song => <span key={song}>{song} | </span>)}
			</div>
		);
	}
}

export default SinglePlaylist;
