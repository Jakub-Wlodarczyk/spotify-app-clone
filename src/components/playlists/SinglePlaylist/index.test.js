import React from 'react';
import { shallow } from 'enzyme';
import SinglePlaylist from './index';

describe('SinglePlaylist component', () => {
	const props = {
		playlist: { name: 'playlist', songs: ['song1', 'song2']}
	};
	it('should match the snapshot', () => {
		const wrapper = shallow(<SinglePlaylist {...props} />);
		expect(wrapper).toMatchSnapshot();
	});
});
