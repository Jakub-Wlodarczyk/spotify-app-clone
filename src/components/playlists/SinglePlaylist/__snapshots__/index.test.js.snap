// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SinglePlaylist component should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SinglePlaylist
    playlist={
      Object {
        "name": "playlist",
        "songs": Array [
          "song1",
          "song2",
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p>
          <b>
            Playlist name:
          </b>
           
          playlist
        </p>,
        <p>
          <b>
            Songs:
          </b>
        </p>,
        Array [
          <span>
            song1
             | 
          </span>,
          <span>
            song2
             | 
          </span>,
        ],
      ],
      "className": "single-song",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <b>
              Playlist name:
            </b>,
            " ",
            "playlist",
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Playlist name:",
            },
            "ref": null,
            "rendered": "Playlist name:",
            "type": "b",
          },
          " ",
          "playlist",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <b>
            Songs:
          </b>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Songs:",
          },
          "ref": null,
          "rendered": "Songs:",
          "type": "b",
        },
        "type": "p",
      },
      Object {
        "instance": null,
        "key": "song1",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "song1",
            " | ",
          ],
        },
        "ref": null,
        "rendered": Array [
          "song1",
          " | ",
        ],
        "type": "span",
      },
      Object {
        "instance": null,
        "key": "song2",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "song2",
            " | ",
          ],
        },
        "ref": null,
        "rendered": Array [
          "song2",
          " | ",
        ],
        "type": "span",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p>
            <b>
              Playlist name:
            </b>
             
            playlist
          </p>,
          <p>
            <b>
              Songs:
            </b>
          </p>,
          Array [
            <span>
              song1
               | 
            </span>,
            <span>
              song2
               | 
            </span>,
          ],
        ],
        "className": "single-song",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <b>
                Playlist name:
              </b>,
              " ",
              "playlist",
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Playlist name:",
              },
              "ref": null,
              "rendered": "Playlist name:",
              "type": "b",
            },
            " ",
            "playlist",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <b>
              Songs:
            </b>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Songs:",
            },
            "ref": null,
            "rendered": "Songs:",
            "type": "b",
          },
          "type": "p",
        },
        Object {
          "instance": null,
          "key": "song1",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "song1",
              " | ",
            ],
          },
          "ref": null,
          "rendered": Array [
            "song1",
            " | ",
          ],
          "type": "span",
        },
        Object {
          "instance": null,
          "key": "song2",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "song2",
              " | ",
            ],
          },
          "ref": null,
          "rendered": Array [
            "song2",
            " | ",
          ],
          "type": "span",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
